doc/auth-handlers.md

```diff
@@ -0,0 +1,34 @@
+## AuthImplementations
+
+If we need to enable auth handlers in Pravega controller, modify the Pravega manifest and ensure that following fields are present in the yaml file.
+
+```
+authImplementations:
+  mountPath: "/opt/pravega/pluginlib"
+  authHandlers:
+  - image: authHandlerImage
+    source: "/some_vendor/data/*"
+```  
+
+User has to provide an image that contains authhandler implementation and the image should have `/bin/sh` enabled.
+
+Pravega performs dynamic implementations of the Authorization/Authentication [API](https://github.com/pravega/pravega/blob/master/documentation/src/docs/auth/auth-plugin.md). Multiple implementations can co-exist and different plugins can be used by different users.
+
+
+Alternatively, user can mention the details in init Containers as well to copy the files to controller or segmentstore pods. For using init containers modify the  Pravega manifest and ensure that following fields are present.
```

Comment:
I think the section about the generic init containers should come outside of the auth-handlers.md because it's not specific to auth.

In fact, one of the very real use cases for the generic init containers it the copying of the bk placement script into the segment store container.    So it should probably be placed in its own .md file.